rholang/examples/vault_demo/1.create_alcie_vault.rho

```diff
@@ -0,0 +1,52 @@
+// A tesetnet admin will create vaults for you if you don't have a friend on-chain already
+
+match (
+    "0000000000000000000000000000000000000000000000000000000000000000".hexToBytes(),
+    "111123jwD7acNy6vfYwzWXAbnSQ8z8Bv24cCkVcVbd6SuP2LtFK7w7",
+    "1111111111111111111111111111111111111111111111111111111111111111".hexToBytes(),
+    "11112comWwT626KfXkvnWRHDwPDm4VefqSr6ijvinhVHGTuXe6wWmk",
+    "2222222222222222222222222222222222222222222222222222222222222222".hexToBytes(),
+    "11112ipm5NY9BqyQ3p3sUx8pdFWV5jFQ5qqMcUzZWe19hwz3USbHBG"
+) {
+  (
+    genesisPubKey,
+    genesisRevAddress,
+    alicePubKey,
+    aliceRevAddress,
+    bobPubKey,
+    bobRevAddress
+  ) => {
+    new
+      rl(`rho:registry:lookup`), revAddress(`rho:rev:address`), RevVaultCh,
+      genesisVaultCh, aliceVaultLockedCh, aliceRevAddressCh, transferDone,
+      stdout(`rho:io:stdout`), log
+    in {
+
+      rl!(`rho:id:1o93uitkrjfubh43jt19owanuezhntag5wh74c6ur5feuotpi73q8z`, *RevVaultCh) |
+      for (@(_, RevVault) <- RevVaultCh) {
+        log!(("1.create_alcie_vault.rho")) |
+
+        @RevVault!("findOrCreateGenesisVault", genesisPubKey, 9000, *genesisVaultCh ) |
+
+        @RevVault!("findOrCreate", alicePubKey, *aliceVaultLockedCh) |
```

Comment:
@ArturGajowy this call needs to be the REV Address, not alicePublicKey.  The REV Address is what they exchange in the real world outside of any RChain software.

Reply:
The exchange does not need to create the wallet to make the transfer, nor does the target wallet need to exist for that. It can be created later based on the pub key used to derive the revaddress, at which point the funds from the exchange will make its way to it. Also, the funds will be deducted from the exchange even if the wallet at the address doesn't exist.

Reply:
perhaps I should ask a different question.  What is the purpose of this "findOrCreate" call? Below, there is a call to:
`genesisVault!("transfer", aliceRevAddress, 1000, *transferDone)`
which should should hide the details of this "findOrCreate" logic.

Reply:
If `transfer` would create the target vault when it doesn't exist, there would be no way of transferring funds across different 'kinds' of vaults, or we'd need to parameterize vaults with a 'vault creation strategy'. We could be targeting a multisig, after all.

You're right that `findOrCreate` is redundant here. I intended to show the balance, but I guess it makes more sense to decouple that into two deploys.